common/src/main/java/org/apache/celeborn/common/protocol/TransportModuleConstants.java

```diff
@@ -24,11 +24,21 @@ public class TransportModuleConstants {
 
   // RPC module used by the application components to communicate with each other
   // This is used only at the application side.
+  // This is interally further split into RPC_LIFECYCLEMANAGER_MODULE and
+  // RPC_APP_CLIENT_MODULE - both of which inherit from RPC_APP_MODULE
+  // So for users, there is only RPC_APP_MODULE
   public static final String RPC_APP_MODULE = "rpc_app";
   // RPC module used to communicate with/between server components
   // This is used both at server (master/worker) and application side.
   public static final String RPC_SERVICE_MODULE = "rpc_service";
 
+  // See RPC_APP_MODULE for details - both RPC_LIFECYCLEMANAGER_MODULE and RPC_APP_CLIENT_MODULE
+  // are internal modules, and transport configs are not expected for these.
+  // For example, auto-ssl requires both RPC_LIFECYCLEMANAGER_MODULE and RPC_APP_CLIENT_MODULE
+  // to be in sync, though it is enabled only in RPC_LIFECYCLEMANAGER_MODULE
+  public static final String RPC_LIFECYCLEMANAGER_MODULE = "rpc_app_lifecyclemanager";
+  public static final String RPC_APP_CLIENT_MODULE = "rpc_app_client";
```

Comment:
What's the difference between `rpc_app_client` and `rpc_app` module? From a user perspective, this can easily be confusing.

Reply:
You are right - which is why this is not exposed to users, it is internal to Celeborn.
Please see the [comment here](https://github.com/apache/celeborn/pull/2471/files#diff-d5cb375c690ff45434efc9ca88e01ff6aa3122ed376656dd060512e955da745cR35) and [this warning ](https://github.com/apache/celeborn/pull/2471/files#diff-d6ae0fde704411e24d619d6034f2a00624c2e579b145fa6f10580c6c6a10adadR90) in case users do end up using it (by looking at the code).

Reply:
@mridulm, IMO, we could distinguish which module or config option is internal in the configuration document. WDYT?

Reply:
Sounds good